### ISSUE TITLE
[Q] Does define_metric allow a summary based on the point in time where another metric performs best?

### ISSUE THREAD

**Issue by AndersMollgaard** [wandb/wandb#4812]
A fair summary of a model should be specified based on the same checkpoint and not different checkpoints where the individual metrics are performing at their best. 

I can see that define_metric has the summary = 'best' option along with a goal = 'minimize' / 'maximize', but as I understand it, I cannot let the goal be based on another metric. Is that so?

For instance, I would like to have a summary including the loss and the accuracy at the point in time, where the accuracy is highest.

**Comment by ArtsiomWB** [comment]
Hi @AndersMollgaard! [Is this](https://docs.wandb.ai/guides/track/log#customize-axes-and-summaries-with-define_metric) somewhat similar to what you are looking for? Our docs have an example of how to customize the summary. 

In the example linked above, `the user is capturing the lowest value of loss and the maximum value of accuracy in the summary, instead of the default summary behavior, which uses the final value from history.`

**Comment by AndersMollgaard** [comment]
> Hi @AndersMollgaard! [Is this](https://docs.wandb.ai/guides/track/log#customize-axes-and-summaries-with-define_metric) somewhat similar to what you are looking for? Our docs have an example of how to customize the summary.

Yeah, I was looking at your documentation, but did not find what I was looking for. If I understand it correctly, then the above will create a summary where the values of the metrics in the summary are potentially evaluated at different points in time during the training. 

I could have a model classifying "dog or cat" which at one point during training says "everything is a dog" and at another point says "everything is a cat". My summary would then say: {"fraction_of_dogs_found": 100%, "fraction_of_cats_found": 100%, "accuracy": ...}.

I would like the summary to contain those two metrics at the point in time where the accuracy was largest. In code it would be something like:

`
wandb.define_metric("accuracy", summary="best", goal="maximize")
wandb.define_metric("fraction_of_dogs_found", summary="value_when_accuracy_best")
wandb.define_metric("fraction_of_cats_found", summary="value_when_accuracy_best")
`

**Comment by ArtsiomWB** [comment]
I have asked around internally and we currently do not have this feature. It does sound like a great feature request I could ticket for you though! I think the workaround for this would be to use Pandas data frame and then log it that way.

**Comment by AndersMollgaard** [comment]
> I have asked around internally and we currently do not have this feature. It does sound like a great feature request I could ticket for you though! I think the workaround for this would be to use Pandas data frame and then log it that way.

Sorry for the late reply. Yes, it would be nice with s feature request 🙂

**Comment by ArtsiomWB** [comment]
No Worries!

Your feature request has been submitted :)

**Comment by alex-hh** [comment]
@ArtsiomWB I'd also like to request this feature, I assume it isn't yet implemented?

**Comment by ArtsiomWB** [comment]
Hi @alex-hh, that is correct. What is your desired workflow here?

**Comment by alex-hh** [comment]
well more or less exactly as above:

```
wandb.define_metric("accuracy", summary="best", goal="maximize")
wandb.define_metric("fraction_of_dogs_found", summary="value_when_accuracy_best")
wandb.define_metric("fraction_of_cats_found", summary="value_when_accuracy_best")
```

**Comment by ArtsiomWB** [comment]
Thank you for elaborating. I have gone ahead and created a new ticket regarding this in order to re-triage the request.
